external/model-preparation-algorithm/mpa_tasks/apis/task.py

```diff
@@ -181,6 +184,34 @@ def _initialize(self, dataset=None, output_model=None, export=False):
                 del self._model_cfg['fp16']
         self._precision = [ModelPrecision.FP32]
 
+        # Add/remove adaptive interval hook
+        if 'use_adaptive_interval' in self._recipe_cfg:
+            if self._recipe_cfg.use_adaptive_interval is True:
+                ada_interval_cfg = self._recipe_cfg.get('adaptive_validation_interval', None)
+                if ada_interval_cfg is None:
+                    self._recipe_cfg.adaptive_validation_interval = dict(max_interval=5)
+            else:
+                self._recipe_cfg.pop('adaptive_validation_interval')
+
+        # Add/remove early stop hook
+        if 'early_stop' in self._recipe_cfg:
+            remove_custom_hook(self._recipe_cfg, 'EarlyStoppingHook')
```

Comment:
Does this mean that this will not use the regular 'EarlyStoppingHook' but only the 'LazyEarlyStoppingHook'?

Reply:
Yes, you are right.